openmetadata-ui/src/main/resources/ui/src/components/BotDetails/BotDetails.component.tsx

```diff
@@ -77,35 +79,36 @@ const BotDetails: FC<BotsDetailProp> = ({
     useState<boolean>(false);
   const [generateToken, setGenerateToken] = useState<boolean>(false);
   const [selectedExpiry, setSelectedExpiry] = useState('7');
+  const [botPermission, setBotPermission] = useState<OperationPermission>(
+    DEFAULT_ENTITY_PERMISSION
+  );
 
   const editAllPermission = useMemo(
-    () =>
-      !isEmpty(permissions) &&
-      checkPermission(Operation.EditAll, ResourceEntity.BOT, permissions),
-    [permissions]
+    () => !isEmpty(botPermission) && botPermission.EditAll,
+    [botPermission]
   );
   const displayNamePermission = useMemo(
-    () =>
-      !isEmpty(permissions) &&
-      checkPermission(
-        Operation.EditDisplayName,
-        ResourceEntity.BOT,
-        permissions
-      ),
-    [permissions]
+    () => !isEmpty(botPermission) && botPermission.EditDisplayName,
```

Comment:
```suggestion
    () => botPermission.EditDisplayName,
```

As we have initialised with some value there's no way it can be empty we can remove that check